### PR TITLE
docs: #1757 clarify TS cleanup policy can be compact,delete

### LINKED ISSUE #1757: clarify TS cleanup policy can be compact,delete
### Issue description

Current doc implies the cleanup policy of only 'compact' is in use with TS. This page explains how TS and compaction interact, but most often the cleanup policy would be 'compact,delete' when compaction is in use (implying the topic has some retention period that's not infinite). So we should talk about "when the topic cleanup policy is compact,delete or compact..."

### Updates to existing documentation

https://docs.redpanda.com/docs/manage/tiered-storage/#set-retention-limits

### Link to Redpanda Slack conversation

https://redpandadata.slack.com/archives/C02BDN76HUK/p1689207269070409?thread_ts=1689193212.170589&cid=C02BDN76HUK

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1757 

This just changes the updates this: 

When you set [cleanup.policy](https://deploy-preview-1760--redpanda-documentation.netlify.app/docs/develop/config-topics/#change-the-cleanup-policy) for a topic to compact, nothing gets deleted from object storage based on retention settings. When set to compact,delete, compacted segments are deleted from object storage based on retention.ms and retention.bytes.

preview
- https://deploy-preview-1760--redpanda-documentation.netlify.app/docs/manage/tiered-storage/#compacted-topics-in-tiered-storage
- https://deploy-preview-1760--redpanda-documentation.netlify.app/docs/manage/kubernetes/tiered-storage/#compacted-topics-in-tiered-storage

### CODE REVIEW

**Comment by mattschumpert** [comment]
I defer to @andrwng  for review here

File: docs/manage/shared/tiered-storage/_set-retention-limits.mdx [hunk 1]
```diff
@@ -12,7 +12,7 @@ Data becomes eligible for deletion from object storage following `retention.ms`
 
 #### Compacted topics in Tiered Storage
 
-When you set <Link to={props.frontmatter.linkRoot + 'develop/config-topics/#change-the-cleanup-policy'}>`cleanup.policy=compact`</Link> for a topic, nothing gets deleted from object storage. 
+When you set <Link to={props.frontmatter.linkRoot + 'develop/config-topics/#change-the-cleanup-policy'}>`cleanup.policy`</Link> for a topic to either `compact,delete` or `compact`, nothing gets deleted from object storage.
```
Comment (andrwng) [inline docs/manage/shared/tiered-storage/_set-retention-limits.mdx:15]: This isn't quite it. How about something like

When you set `cleanup.policy` for a topic to `compact`, nothing gets deleted from object storage based on retention settings. When set to `compact,delete`, compacted segments are deleted from object storage based on `retention.ms` and `retention.bytes`.

**Review by mattschumpert — APPROVED** [review]

**Review by andrwng — APPROVED** [review]

**Comment by micheleRP** [comment]
> Should we have a link to this section here? https://docs.redpanda.com/docs/reference/topic-properties/#cleanuppolicy

Yes, that's a better link! I'll change from linking [here](https://deploy-preview-1760--redpanda-documentation.netlify.app/docs/develop/config-topics/#change-the-cleanup-policy) to [there](https://docs.redpanda.com/docs/reference/topic-properties/#cleanuppolicy).

**Comment by JakeSCahill** [comment]
> > Should we have a link to this section here? https://docs.redpanda.com/docs/reference/topic-properties/#cleanuppolicy
> 
> Yes, that's a better link! I'll change from linking [here](https://deploy-preview-1760--redpanda-documentation.netlify.app/docs/develop/config-topics/#change-the-cleanup-policy) to [there](https://docs.redpanda.com/docs/reference/topic-properties/#cleanuppolicy).

On that reference page under 'related topics', we should add a link to the 'compacted topics' section too.

**Pull request merged**